tests/integration/conftest.py

```diff
@@ -404,3 +420,67 @@ def run_job_cmd(
         return polled_tasks
 
     return _disable_polling
+
+
+@pytest.fixture
+def _setup_validate_cli():
+    """Provides CLI arguments for cylc validate."""
+    def _inner(opts=None):
+        setup = SimpleNamespace()
+        setup.parser = validate_gop()
+        setup.opts = setup.parser.get_default_values()
+        setup.opts.templatevars = []
+        setup.opts.templatevars_file = []
+        if setup.opts is not None:
+            for key, val in opts.items():
+                setattr(setup.opts, key, val)
+        return setup
+    yield _inner
+
+
+@pytest.fixture
+def _source_workflow(run_dir, flow_src):
+    """Create a workflow in source directory and install it.
+
+    Useful for testing interaction between installed and source directories.
+
+    Args:
+        cfg: Can be passed a config dictionary.
+
+    Yields:
+        A namespace object with properties:
+            src: source location.
+            opts: options set for cylc install.
+
+    """
+    def _inner(cfg=None):
+        # Object to store info about the setup.
+        setup = SimpleNamespace()
+
+        # Create a workflow source:
+        if cfg is None:
+            setup.src = flow_src({
+                'scheduler': {
+                    'allow implicit tasks': True
+                },
+                'scheduling': {
+                    'initial cycle point': '1854',
+                    'graph': {
+                        'R1': 'foo'
+                    }
+                }
+            })
+        else:
+            setup.src = flow_src(cfg)
```

Comment:
For a "default" config tests can use `one_conf` which is a little more explicit.

E.G. here's how we might create a simple workflow where we don't care about the config:

```python
def test_something(flow, one_conf):
    id_ = flow(one_conf)
```

Suggest making `src_flow` work the same way (i.e. making `cfg` mandatory).